libcore/FileUtils.vala

```diff
@@ -21,14 +21,15 @@ namespace Files.FileUtils {
      **/
     const string RESERVED_CHARS = (GLib.Uri.RESERVED_CHARS_GENERIC_DELIMITERS +
                                    GLib.Uri.RESERVED_CHARS_SUBCOMPONENT_DELIMITERS + " ");
+
     public GLib.List<GLib.File> files_from_uris (string uris) {
         var result = new GLib.List<GLib.File> ();
         var uri_list = GLib.Uri.list_extract_uris (uris);
-        string unquoted_uri;
         foreach (unowned string uri in uri_list) {
             try {
-                unquoted_uri = Shell.unquote (uri); // Extracted uri may be quoted
-                result.append (GLib.File.new_for_uri (unquoted_uri));
+                var unquoted_uri = Shell.unquote (uri); // Extracted uri may be quoted
+                var escaped_uri = escape_uri (unquoted_uri); // Uris from clipboard or DNDHandler are unescaped/sanitized
```

Comment:
This function handles uris, they must be escaped if they get to here. And when checking with prints they appear identical:
<img width="1075" height="122" alt="image" src="https://github.com/user-attachments/assets/4fd43d2b-b4da-402a-9bd3-051a5f7fa20a" />

Reply:
Yes, URIs **must** escape special characters as specified by URI syntax -- https://en.wikipedia.org/wiki/Uniform_Resource_Identifier#Syntax. If it's not escaped the URI is invalid. I don't think we should try to fix issues in other applications that use URI incorrectly.